homeassistant/components/obihai/button.py

```diff
@@ -0,0 +1,59 @@
+"""Obihai button module."""
+
+from __future__ import annotations
+
+from pyobihai import PyObihai
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_platform
+
+from .connectivity import ObihaiConnection
+from .const import OBIHAI
+
+BUTTON_DESCRIPTION = ButtonEntityDescription(
+    key="reboot",
+    name=f"{OBIHAI} Reboot",
```

Comment:
This seems incorrect.

`Reboot` -> `reboot`. The prefixing should not be needed?

Reply:
Leaving out the prefix results in an entity named "reboot" which IMHO is confusing and undesirable.

![image](https://user-images.githubusercontent.com/21226768/222773300-fd36cde3-9782-433e-a671-9fc03ab90f6e.png)

I do see that if I use attributes instead of setting entity_description (comment below) that this does work as you indicated, so I'll use that approach.

Reply:
The problem seems the missing DeviceInfo that causes that, if you add device info it will grouped in the device.